aeppl/mixture.py

```diff
@@ -248,14 +250,20 @@ def mixture_replace(fgraph, node):
     From these terms, new terms ``Z_rv[i] = mixture_comps[i][i == I_rv]`` are
     created for each ``i`` in ``enumerate(mixture_comps)``.
     """
-
     rv_map_feature = getattr(fgraph, "preserve_rv_mappings", None)
 
     if rv_map_feature is None:
         return None  # pragma: no cover
 
     old_mixture_rv = node.default_output()
 
+    if isinstance(node.op, Elemwise):
+        if not isinstance(node.op.scalar_op, Switch):
```

Comment:
A distinct `local_optimizer` would essentially move this check up the stack into `LocalOptGroup.transform`.